src/IIIFPresentation/API.Tests/Integration/GetManifestTests.cs

```diff
@@ -88,4 +100,128 @@ public async Task Get_IiifManifest_Hierarchical_ReturnsManifestFromS3()
+
+        // Assert
+        response.StatusCode.Should().Be(HttpStatusCode.Accepted);
+        response.Headers.Should().ContainKey(HeaderNames.ETag);
+        response.Headers.Vary.Should().HaveCount(2);
+        manifest.Should().NotBeNull();
+        manifest!.Type.Should().Be("Manifest");
+        manifest.Id.Should().Be($"http://localhost/1/manifests/{id}", "requested by flat URI");
+        manifest.Items.Should().HaveCount(3, "the test content contains 3 children");
+        manifest.FlatId.Should().Be(id);
+        manifest.PublicId.Should().Be($"http://localhost/1/{id}", "iiif-manifest is slug and under root");
+    }
+    
+    [Fact]
+    public async Task Get_IiifManifest_Hierarchical_ReturnsNotFoundWhenIngesting()
+    {
+        // Arrange and Act
+        var id = nameof(Get_IiifManifest_Hierarchical_ReturnsNotFoundWhenIngesting);
+        
+        // Arrange and Act
+        await dbContext.Manifests.AddAsync(GenerateManifestRecord(id, 2));
+        
+        await amazonS3.PutObjectAsync(new()
+        {
+            BucketName = LocalStackFixture.StorageBucketName,
+            Key = $"1/manifests/{id}",
+            ContentBody = TestContent.ManifestJson
+        });
+        
+        await dbContext.SaveChangesAsync();
+        
+        var response = await httpClient.GetAsync($"1/{id}");
+        // Assert
+        response.StatusCode.Should().Be(HttpStatusCode.NotFound);
+        response.Headers.Vary.Should().HaveCount(2);
+    }
+
+    [Fact]
+    public async Task Get_IiifManifest_Hierarchical_ReturnsOkWhenIngestingButHasIngestedBefore()
+    {
+        // Arrange and Act
+        var id = nameof(Get_IiifManifest_Hierarchical_ReturnsOkWhenIngestingButHasIngestedBefore);
+        
+        // Arrange and Act
+        await dbContext.Manifests.AddAsync(GenerateManifestRecord(id, 3, true));
+        
+        await amazonS3.PutObjectAsync(new()
+        {
+            BucketName = LocalStackFixture.StorageBucketName,
+            Key = $"1/manifests/{id}",
+            ContentBody = TestContent.ManifestJson
+        });
+        
+        await dbContext.SaveChangesAsync();
+        
+        var response = await httpClient.GetAsync($"1/{id}");
+        
+        var manifest = await response.ReadAsPresentationJsonAsync<PresentationManifest>();
+        
+        // Assert
+        response.StatusCode.Should().Be(HttpStatusCode.OK);
+        response.Headers.Should().ContainKey(HeaderNames.ETag);
+        response.Headers.Vary.Should().HaveCount(2);
+        manifest.Should().NotBeNull();
+        manifest!.Type.Should().Be("Manifest");
+        manifest.Id.Should().Be($"http://localhost/1/{id}", "requested by hierarchical URI");
+        manifest.Items.Should().HaveCount(3, "the test content contains 3 children");
+    }
+
+    private static Models.Database.Collections.Manifest GenerateManifestRecord(string id, int batchId, bool ingested = false)
```

Comment:
Is it worth extending the `AddTestManifest` method, or creating an overload, so that it's widely usable moving forward?